src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  be4toi,
+  be8toi,
+  bytesToStr,
+  concat,
+  itobe4,
+  itobe8,
+  strToBytes,
+} from "../../utils/bytes";
+
+/**
+ * Create a new _Atom_ (isobmff box).
+ * @param {string} name - The box name (e.g. sidx, moov, pssh etc.)
+ * @param {Uint8Array} buff - The box's content
+ */
+function Atom(name : string, buff : Uint8Array) : Uint8Array {
+  const len = buff.length + 8;
+  return concat(itobe4(len), strToBytes(name), buff);
+}
+
+export default function patchSegmentWithTimeOffset(
+      _data: Uint8Array,
+      _offset: number,
+      _lmsg?: boolean
+  ) {
+      let sizeChange = 0;
+      const segmentData = _data;
+      const topLevelBoxesToParse = ["moov", "styp", "sidx", "moof", "mdat"];
+      const compositeBoxesToParse = ["trak", "moov", "moof", "traf"];
+      const timeOffset = _offset;
+      const lmsg = _lmsg || false;
+
+  /**
+   * Get size and boxtype from current box
+   * @param {Uint8Array} data
+   */
+  function getBoxInfos(boxData: Uint8Array) {
+    const size = be4toi(boxData, 0);
+    const boxtype = bytesToStr(boxData.subarray(4, 8));
+    return {
+      size,
+      boxtype,
+    };
+  }
+
+  /**
+   * Triggers box processing
+   */
+  function patchBoxes() {
+    let output = new Uint8Array(0);
+    let pos = 0;
+    while (pos < segmentData.length) {
+      const { size, boxtype } = getBoxInfos(segmentData.subarray(pos, pos + 8));
+      const boxdata = segmentData.subarray(pos, pos + size);
+      const concatData = (topLevelBoxesToParse
+        .indexOf(boxtype) >= 0) ?
+          patchBox(boxtype, boxdata, output.length) :
+          boxdata;
+      output = concat(output, concatData);
+      pos += size;
+    }
+    return output;
+  }
+
+  /**
+   * Process specific box and its children
+   * @param {string} boxtype
+   * @param {Uint8Array} data
+   * @param {number} filePos
+   * @param {string} _path
+   */
+  function patchBox(
+    boxtype: string,
+    boxSpecificData: Uint8Array,
+    filePos: number,
+    _path?: string
+  ): Uint8Array {
+    let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWLjl6y--pl6AHs2a-uR&open=AWLjl6y--pl6AHs2a-uR)

[ISSUE_KEY]: AWLjl6y--pl6AHs2a-uR